src/scheduler.cc

```diff
@@ -700,50 +692,54 @@ void SchedulerService::deallocate_object(ObjRef canonical_objref) {
-      objstores_[objstoreid].objstore_stub->DeallocateObject(&context, request, &reply);
+      (*objstores)[objstoreid].objstore_stub->DeallocateObject(&context, request, &reply);
     }
-    objtable_[canonical_objref].clear();
+    locations.clear();
   }
-  decrement_ref_count(contained_objrefs_[canonical_objref]);
+  decrement_ref_count((*contained_objrefs_.unsafe_get())[canonical_objref]);
 }
 
 void SchedulerService::increment_ref_count(const std::vector<ObjRef> &objrefs) {
   // increment_ref_count assumes that reference_counts_lock_ has been acquired already
   for (int i = 0; i < objrefs.size(); ++i) {
     ObjRef objref = objrefs[i];
-    RAY_CHECK_NEQ(reference_counts_[objref], DEALLOCATED, "Attempting to increment the reference count for objref " << objref << ", but this object appears to have been deallocated already.");
-    reference_counts_[objref] += 1;
-    RAY_LOG(RAY_REFCOUNT, "Incremented ref count for objref " << objref <<". New reference count is " << reference_counts_[objref]);
+    RAY_CHECK_NEQ((*reference_counts_.unsafe_get())[objref], DEALLOCATED, "Attempting to increment the reference count for objref " << objref << ", but this object appears to have been deallocated already.");
+    (*reference_counts_.unsafe_get())[objref] += 1;
+    RAY_LOG(RAY_REFCOUNT, "Incremented ref count for objref " << objref <<". New reference count is " << (*reference_counts_.unsafe_get())[objref]);
   }
 }
 
 void SchedulerService::decrement_ref_count(const std::vector<ObjRef> &objrefs) {
-  // decrement_ref_count assumes that reference_counts_lock_ has been acquired already
+  // decrement_ref_count assumes that reference_counts_lock_ and
+  // contained_objrefs_lock_ have been acquired already. contained_objrefs_lock_
+  // is needed inside of deallocate_object
   for (int i = 0; i < objrefs.size(); ++i) {
     ObjRef objref = objrefs[i];
-    RAY_CHECK_NEQ(reference_counts_[objref], DEALLOCATED, "Attempting to decrement the reference count for objref " << objref << ", but this object appears to have been deallocated already.");
-    RAY_CHECK_NEQ(reference_counts_[objref], 0, "Attempting to decrement the reference count for objref " << objref << ", but the reference count for this object is already 0.");
-    reference_counts_[objref] -= 1;
-    RAY_LOG(RAY_REFCOUNT, "Decremented ref count for objref " << objref << ". New reference count is " << reference_counts_[objref]);
+    RAY_CHECK_NEQ((*reference_counts_.unsafe_get())[objref], DEALLOCATED, "Attempting to decrement the reference count for objref " << objref << ", but this object appears to have been deallocated already.");
+    RAY_CHECK_NEQ((*reference_counts_.unsafe_get())[objref], 0, "Attempting to decrement the reference count for objref " << objref << ", but the reference count for this object is already 0.");
+    (*reference_counts_.unsafe_get())[objref] -= 1;
```

Comment:
Store the output of `unsafe_get` in a variable and reuse that when you need it, instead of calling it twice. That way when you run into a bug later and start looking for calls to `unsafe_get`, you won't have to examine and/or change every single one separately.